src/Parser/Monk/Brewmaster/Modules/Normalizers/IronskinBrew.js

```diff
@@ -0,0 +1,97 @@
+import SPELLS from 'common/SPELLS';
+import EventsNormalizer from 'Parser/Core/EventsNormalizer';
+
+const debug = false;
+
+/**
+ * This class normalizes the event stream for the bug specifically with
+ * ISB: if the fight starts with ISB up, and the combatant never lets it
+ * drop, then there are no buff events for it. ApplyBuff doesn't
+ * normalize correctly in this case because no buff-related
+ * events fire for it ever, only casts.
+ */
+class IronskinBrew extends EventsNormalizer {
+  // stolen from ApplyBuff
+  _getFirstEventIndex(events) {
+    for (let i = 0; i < events.length; i += 1) {
+      const event = events[i];
+      if (event.type !== 'combatantinfo') {
+        return i;
+      }
+    }
+    throw new Error('Fight doesn\'t have a first event, something must have gone wrong.');
+  }
+
+  _combatantInfoEvents = {};
+  initialize(combatants) {
+    this._combatantInfoEvents = combatants;
+  }
+
+  normalize(events) {
+    const isbCasts = {};
+    const isbBuffs = {};
+    let isbBuffAbility;
+    for(let i = 0; i < events.length; i++) {
+      const event = events[i];
+      const sourceID = event.sourceID;
+      const targetID = event.targetID;
+      const spellId = event.ability ? event.ability.guid : null;
+
+      if(event.type === "cast" && SPELLS.IRONSKIN_BREW.id === spellId) {
```

Comment:
Please don't use [Yoda conditions](https://en.wikipedia.org/wiki/Yoda_conditions), they're often harder to read and more importantly inconsistent with the rest of the codebase.